src/lib/fcitx/instance.h

```diff
@@ -73,6 +73,13 @@ enum class EventWatcherPhase {
 
 struct FCITXCORE_EXPORT InstanceQuietQuit : public std::exception {};
 
+enum class DeviceMode {
+
+    PC = 0,
```

Comment:
naming is a little bit confusing here.

Reply:
Also what's the difference between device mode and input method mode? why it needs to be two different enum?

Reply:
Input method mode is used to differentiate virtual keyboard and physical keyboard.
Device mode is used to differentiate if the device is in PC mode or Tablet mode.

Reply:
So far, device mode is only used to not hide virtual keyboard when focus is out on PC if there is an available physical keyboard.
If there is no available physical keyboard or device mode is tablet, the virtual keyboard is automatically closed when focus is out.